huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_mysql_binlog_test.go

```diff
@@ -0,0 +1,159 @@
+	mysqlBinlogOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+	mysqlBinlogResp, err := mysqlBinlogClient.Request("GET", mysqlBinlogPath, &mysqlBinlogOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDS Mysql binlog")
+	}
+	mysqlBinlogRespBody, err := utils.FlattenResponse(mysqlBinlogResp)
+	if err != nil {
+		return nil, err
+	}
+	binlog := utils.PathSearch("binlog_retention_hours", mysqlBinlogRespBody, nil)
+	if binlog == nil {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	return mysqlBinlogRespBody, nil
+}
+
+func TestAccMysqlBinlog_basic(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_rds_mysql_binlog.test"
+	dbPwd := fmt.Sprintf("%s%s%d", acctest.RandString(5),
+		acctest.RandStringFromCharSet(2, "!#%^*"), acctest.RandIntRange(10, 99))
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getMysqlBinlogResourceFunc,
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testMysqlBinlog_basic(name, dbPwd),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "instance_id",
+						"huaweicloud_rds_instance.test", "id"),
+					resource.TestCheckResourceAttr(rName, "binlog_retention_hours", "6"),
+				),
+			},
+			{
+				Config: testMysqlBinlog_basic_update(name, dbPwd),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "instance_id",
+						"huaweicloud_rds_instance.test", "id"),
+					resource.TestCheckResourceAttr(rName, "binlog_retention_hours", "8"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccRdsInstance_mysql(name, pwd string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_rds_flavors" "test" {
+db_type       = "MySQL"
+db_version    = "8.0"
+instance_mode = "single"
+group_type    = "dedicated"
+}
+
+resource "huaweicloud_rds_instance" "test" {
+name              = "%[2]s"
+flavor            = data.huaweicloud_rds_flavors.test.flavors[0].name
+security_group_id = data.huaweicloud_networking_secgroup.test.id
+subnet_id         = data.huaweicloud_vpc_subnet.test.id
+vpc_id            = data.huaweicloud_vpc.test.id
+availability_zone = slice(sort(data.huaweicloud_rds_flavors.test.flavors[0].availability_zones), 0, 1)
+ssl_enable        = true
+
+	db {
+	  password = "%[3]s"
+	  type     = "MySQL"
+	  version  = "8.0"
+	  port     = 3306
+	}
+
+	volume {
+	  type = "CLOUDSSD"
+	  size = 40
+	}
+
```

Comment:
invalid format, pls remove all tab characters and using two spaces instead.

Reply:
done